prov/util/src/util_cq.c

```diff
@@ -429,7 +429,10 @@ int ofi_cq_cleanup(struct util_cq *cq)
 
 	ofi_atomic_dec32(&cq->domain->ref);
 	util_comp_cirq_free(cq->cirq);
-	ofi_mutex_destroy(&cq->mutex);
+	if (cq->lock_type == OFI_LOCK_MUTEX)
```

Comment:
Would it be better to put `lock_type`, `mutex`, `spinlock` into 1 data structure (like `ofi_lock`), and have utility functions liks `ofi_lock_init` and `ofi_lock_destroy`?